sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/implementation/OkHttpResponseBinaryData.java

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.generic.core.http.okhttp.implementation;
+
+import com.generic.core.models.BinaryData;
+import com.generic.core.models.ByteArrayBinaryData;
+import com.generic.core.models.TypeReference;
+import com.generic.core.util.serializer.ObjectSerializer;
+import okhttp3.ResponseBody;
+import okio.Okio;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UncheckedIOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.Channels;
+import java.nio.channels.WritableByteChannel;
+import java.util.Objects;
+
+/**
+ * {@link BinaryData} based on {@link okhttp3.Response}.
+ */
+public class OkHttpResponseBinaryData extends BinaryData {
```

Comment:
is it significantly different from `InputStreamBinaryData` ? Perhaps we can use it for the time being and keep the optimizations in mind?

Reply:
To me, this results in `BinaryData` being the end-to-end handler of the HTTP response body. And unlike `InputStreamBinaryData` this will close the underlying connection when the body is closed, either manually or using try-with-resources.